From 257c95a74f53f2cf01be04352e9ca36b226f2afc Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:43:32 +0800
Subject: [PATCH 2/3] GSI Android 17 compatibility:
 0002-Prepare-for-the-future-making-bpfloader.rs-fails-all.patch

Change-Id: I4363d42fb2e69413d9b3454ea555b5dfd2055215
---
 loader/bpfloader.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/loader/bpfloader.rs b/loader/bpfloader.rs
index f23d882..4a8d0ee 100644
--- a/loader/bpfloader.rs
+++ b/loader/bpfloader.rs
@@ -733,9 +733,9 @@ fn load_libbpf_progs() {
     for file_desc in file_vec {
         if let Err(e) = libbpf_worker(&file_desc) {
             if file_desc.critical {
-                panic!("Error when loading {0}: {e}", file_desc.filename);
+                error!("'critical' Error when loading {0}: {e}\n", file_desc.filename);
             } else {
-                error!("Error when loading {0}: {e}", file_desc.filename);
+                error!("Error when loading {0}: {e}\n", file_desc.filename);
             }
         };
     }
